### ISSUE TITLE
[1.5][MacOS] TypeError: Cannot read property 'get' of undefined

### ISSUE THREAD

**Issue by rlora** [meteor/meteor#8902]
I was running two applications using `Meteor 1.5` then I've run an update to dependencies and I'm not been able to start the apps. Error is:

```
=> Exited with code: 1
W20170712-18:38:41.572(-4)? (STDERR) /Users/rolando/.meteor/packages/meteor-tool/.1.5.0.giur44++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20170712-18:38:41.573(-4)? (STDERR) 						throw(ex);
W20170712-18:38:41.574(-4)? (STDERR) 						^
W20170712-18:38:41.576(-4)? (STDERR) 
W20170712-18:38:41.577(-4)? (STDERR) TypeError: Cannot read property 'get' of undefined
W20170712-18:38:41.578(-4)? (STDERR)     at withoutInvocation (packages/meteor.js:443:33)
W20170712-18:38:41.578(-4)? (STDERR)     at bindAndCatch (packages/meteor.js:452:33)
W20170712-18:38:41.579(-4)? (STDERR)     at Object._.extend.defer (packages/meteor.js:514:26)
W20170712-18:38:41.579(-4)? (STDERR)     at [object Object].MongoConnection.tail (packages/mongo/mongo_driver.js:1170:10)
W20170712-18:38:41.580(-4)? (STDERR)     at [object Object]._.extend._startTailing (packages/mongo/oplog_tailing.js:239:50)
W20170712-18:38:41.581(-4)? (STDERR)     at new OplogHandle (packages/mongo/oplog_tailing.js:76:8)
W20170712-18:38:41.581(-4)? (STDERR)     at new MongoConnection (packages/mongo/mongo_driver.js:214:25)
W20170712-18:38:41.582(-4)? (STDERR)     at new MongoInternals.RemoteCollectionDriver (packages/mongo/remote_collection_driver.js:4:16)
W20170712-18:38:41.582(-4)? (STDERR)     at Object.<anonymous> (packages/mongo/remote_collection_driver.js:38:10)
W20170712-18:38:41.583(-4)? (STDERR)     at Object.defaultRemoteCollectionDriver (packages/underscore.js:784:19)
```

If I **revert** the update the application works as expected.

**Bootstraping a new app with the latest packages also works.**

I think is something related with my apps and suspect `meteor@1.7.0` is the root cause. Will find out more. In the mean time posting the issue in case someone bump into something similar.

```
--- a/.meteor/versions
+++ b/.meteor/versions
@@ -4 +4 @@ accounts-oauth@1.1.15
-accounts-password@1.3.6
+accounts-password@1.3.7
@@ -10 +10 @@ autoupdate@1.3.12
-babel-compiler@6.19.2
+babel-compiler@6.19.4
@@ -17 +17 @@ blaze-tools@1.0.10
-boilerplate-generator@1.1.0
+boilerplate-generator@1.1.1
@@ -27 +27 @@ ddp-client@1.3.4
-ddp-common@1.2.8
+ddp-common@1.2.9
@@ -33 +33 @@ dynamic-import@0.1.1
-ecmascript@0.8.0
+ecmascript@0.8.1
@@ -35 +35 @@ ecmascript-runtime@0.4.1
-ecmascript-runtime-client@0.4.1
+ecmascript-runtime-client@0.4.2
@@ -69 +69 @@ mdg:validation-error@0.5.1
-meteor@1.6.1
+meteor@1.7.0
@@ -75,2 +75,2 @@ minifier-css@1.2.16
-minifier-js@2.1.0
-minimongo@1.2.0
+minifier-js@2.1.1
+minimongo@1.2.1
@@ -79 +79 @@ mobile-status-bar@1.0.14
-modules@0.9.1
+modules@0.9.2
@@ -127 +127 @@ nickel:weather@0.1.0
-npm-bcrypt@0.9.2
+npm-bcrypt@0.9.3
@@ -145 +145 @@ sha@1.0.9
-shell-server@0.2.3
+shell-server@0.2.4
@@ -150 +150 @@ standard-minifier-css@1.3.4
-standard-minifier-js@2.1.0
+standard-minifier-js@2.1.1
@@ -160 +160 @@ url@1.1.0
-webapp@1.3.16
+webapp@1.3.17
(END)
```

**Comment by coagmano** [comment]
Me too, I'm guessing it has to do with Meteor 1.5.1 being released on github / packages updated, but the cli hasn't picked up the new release yet.

I'm trying updating to 1.5.1 manually with `meteor update --release 1.5.1` to see if that fixes it, otherwise will revert

**Comment by coagmano** [comment]
Update: `meteor update --release 1.5.1` fixed it!

**Comment by rlora** [comment]
Thank you @coagmano! I confirm manually updating fixes the issue.

**Comment by hajush** [comment]
I'm seeing this same issue after doing an update from Meteor 1.3 to 1.5 on macos 10.12.5. The default "meteor update" took me to 1.5, not 1.5.1. I received the same error after doing the update and also updating my mongodb version. Right after I explicitly ran "meteor update --release 1.5.1", the app started and was working without a hitch.

**Comment by copleykj** [comment]
I think this should be re-opened. I've encountered this same issue and while updating a project to 1.5.1 fixes the issue for that project, I'm getting this same issue now when creating new projects with prior releases such as 1.4.4.3

**Comment by rlora** [comment]
Reopening since it seems to be more than a temporal issue.

**Comment by josiahres** [comment]
I had the same issue. After running meteor update --release 1.5.1 this error is no more! 👍

**Comment by mvandenbrink** [comment]
I had the same issue. Running on Mac.

Got this error in my browser:
Uncaught TypeError: Cannot read property 'get' of undefined in polyfill:3

After update to 1.5.1 and a clean npm install the problem was solved.

**Comment by steph643** [comment]
Same for me: updating to 1.5.1 solved the issue.

**Comment by hwillson** [comment]
Since 1.5.1 is now the default (recommended) version, and upgrading to 1.5.1 fixes this issue, I'll close this off. Thanks all!

**Comment by a4xrbj1** [comment]
Please open again. We see some weird behavior in WebStorm when tracing our app (which worked fine with 1.4.3.1 that only happens now after upgrading to 1.5.1. Please note that our app is ONLY a backend app. The frontend app (separate app) runs without problems.

Following weird behavior shows up:

- all variables in a function are undefined, except globally defined ones
- all parameters in a function are undefined, except globally defined ones
- it jumps from a CONTINUE statement up to the next higher IF clause (instead of the FOR loop) and then on a step-in command jump 180 lines down into a much deeper nested CONTINUE statement.

I've made a gif of that behavior, see attached (it jumps from line 81 (CONTINUE) to line 72 (IF clause) to line 261 (another CONTINUE)

I've tested it on 1.5, same behavior. Now downgraded back to 1.4.3.1 and I'm stuck with this issue:

> TypeError: Cannot read property 'get' of undefined
> at withoutInvocation (packages/meteor.js:443:33)

We did many code changes in the meantime so rolling back isn't an option

Thanks!
![meteorinstall server](https://user-images.githubusercontent.com/13525267/28329895-2ad2c0ec-6c1e-11e7-9db8-8854f427a27c.gif)

**Comment by coagmano** [comment]
@a4xrbj1 That looks like a separate (and serious) issue and should be posted as such
https://github.com/meteor/meteor/issues/new

**Comment by abernix** [comment]
@a4xrbj1 As suggested by @coagmano, please do open a new issue with your findings and a reproduction.  While there may be some issues related to #8611 (which was intended to be fixed in Meteor 1.5.1) I'd ask that you try to reproduce the problem outside of WebStorm, in Chrome or Safari Dev tools directly, in addition to trying Meteor 1.5 and some of the versions on the 1.4.4.x-series to try to narrow down the problem.  Currently, as you've presented it, it's hard to determine exactly where the problem might lie as there are several versions between 1.4.3.1 and 1.5.1.

Also, I'll point out that when downgrading to a previous version of Meteor, it's quite important to use version control to ensure that the `.meteor/versions` file is restored to its previous state since some packages might have more flexible version constraints than others.

**Comment by sdarnell** [comment]
@a4xrbj1 there's a similar issue where debugging/srcmaps in typescript projects stopped working reliably on the switch from meteor 1.4.4.3 to 1.5. https://github.com/barbatus/typescript/issues/56
Does you project use typescript?

Also, in https://github.com/meteor/meteor/issues/8611 some people found that they needed to clear out their babel cache, I'd suggest you try that too.  `rm -rf ~/.babel-cache`
